src/test/java/entities/validity_checkers/PunctuationValidityCheckerRegularTests.java

```diff
@@ -0,0 +1,150 @@
+package entities.validity_checkers;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class PunctuationValidityCheckerRegularTests {
```

Comment:
Keep some of the unique tests separate, but also consider adding a testValidPunctsMany and testInvalidPunctsMany which would contain long arrays of valid and invalid combos respectively and loop through them and assert null or not null, without checking the modification. List in the doc string the array contents for the puncts you are checking